packages/components/bucket-policy/bucket-policy-body.tsx

```diff
@@ -0,0 +1,442 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import classNames from 'classnames';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import {
+  Checkbox,
+  Divider,
+  Form,
+  FormGroup,
+  TextInput,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Button,
+  Radio,
+} from '@patternfly/react-core';
+import { DataResourceType, EDIT_DATA_RESOURCES } from '../../constants';
+import {
+  NooBaaNamespaceStoreModel,
+  ProjectModel,
+  NooBaaObjectBucketClaimModel,
+} from '../../models';
+import { NamespaceStoreKind, K8sResourceKind } from '../../types';
+import { referenceForModel } from '../../utils';
+import { GenericDropdown } from '../../utils/dropdown/GenericDropdown';
+import { StatusBox } from '../../utils/generics/status-box';
+import { useDeepCompareMemoize } from '../../utils/hooks/deep-compare-memoize';
+import { LaunchModal } from '../../utils/modals/modalLauncher';
+import { getName, getUID, getNamespace } from '../../utils/selectors/k8s';
+import {
+  ReplicationOBC,
+  BucketPolicyState,
+  BucketPolicyAction,
+  BucketPolicyActionType,
+} from './state';
+
+const onChangeMulti: onChangeMultiSelectProps =
+  (data, dispatch) => (selection) => {
+    const alreadyIncluded = data.includes(selection);
+    const payload = alreadyIncluded
+      ? data.filter((resource) => resource !== selection)
+      : [...data, selection];
+    dispatch({
+      type: BucketPolicyActionType.SET_READ_MULTI,
+      payload,
+    });
+  };
```

Comment:
is it ok to use the state var on external function? I am getting a doubt here

Reply:
It should be okay IMO... we are passing them when calling `onChangeMulti` and also `onChangeMulti` is not added to any dependency list so even if its reference changes it should not be an issue...